File: backend/Models/Appointment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models
{
    public class Appointment
    {
        public int Id { get; set; }

        [Required]
        public string PatientId { get; set; } = string.Empty;

        [ForeignKey("PatientId")]
        public User Patient { get; set; } = null!;

        [Required]
        public int AvailabilityId { get; set; }

        [ForeignKey("AvailabilityId")]
        public Availability Availability { get; set; } = null!;

        // Stores comma-separated list of tasks (e.g., "Medication, Vitals, Exercises")
        [Required(ErrorMessage = "At least one task is required.")]
        public string Tasks { get; set; } = string.Empty;

        [DataType(DataType.Time)]
        public TimeSpan StartTime { get; set; }

        [DataType(DataType.Time)]
        public TimeSpan EndTime { get; set; }

        [RegularExpression(@"^(Booked|InProgress|Completed|Cancelled)$", ErrorMessage = "Invalid status value.")]
        public string Status { get; set; } = "Booked";
    }
}

File: frontend/src/appointments/types/appointment.ts
// Appointment model based on backend.DTOs.AppointmentDto
export interface Appointment {
  id?: number;
  patientId: string;
  patientName?: string;
  patientAddress?: string;
  availabilityId: number;
  personnelId?: string;
  personnelName?: string;
  date?: string;
  availabilityNotes?: string;
  // Stores comma-separated list of tasks (e.g., "Medication, Vitals, Exercises")
  tasks: string;
  startTime: string; // TimeSpan represented as string "HH:mm:ss" or "HH:mm"
  endTime: string;
  status: 'Booked' | 'InProgress' | 'Completed' | 'NotCompleted' | 'Cancelled';
  patientSupabaseProfileId?: string | null;
  // Summary of the actual visit for this appointment, if one was started.
  visitId?: number | null;
  visitStatus?: 'Active' | 'Completed' | 'Incomplete' | 'Cancelled' | null;
  visitType?: 'Physical' | 'Digital' | null;
}

// Appointment create DTO
export interface AppointmentCreateDto {
  patientId?: string;
  availabilityId: number;
  // Stores comma-separated list of tasks (e.g., "Medication, Vitals, Exercises")
  tasks: string;
  startTime: string;
  endTime: string;
  status?: string;
}

// Appointment summary for dashboard display
export interface AppointmentSummary {
  id: number;
  patientName: string;
  personnelName: string;
  // Stores comma-separated list of tasks (e.g., "Medication, Vitals, Exercises")
  tasks: string;
  date: string; // ISO date string
  startTime: string;
  endTime: string;
  status: string;
  formattedDateTime?: string;
  availabilityNotes?: string;
  visitId?: number | null;
  visitStatus?: string | null;
  visitType?: string | null;
}
